parsers/BO.py

```diff
@@ -54,76 +66,73 @@ def fetch_production(
             timestamp = now
 
         if target_datetime is not None and hour < 24:
```

Comment:
This whole timestamp handling is hard to read. Maybe extract to a fx?

Reply:
Yes. After testing it it also turns out it had two flaws:
- hour 23 was always skipped
- when using target_datetime the timezone information was not provided, so the returned data was wrongly in UTC

This should be fixed now. The part about skipping if we are in the future is already handled in the fetch_production function as part of the `if row.total is None or None in modes_extracted` check.